main.tf

```diff
@@ -130,15 +130,43 @@ resource "kubernetes_network_policy_v1" "this" {
 
     ingress {
       dynamic "from" {
-        for_each = each.value == "isolated" ? { "isolated" : true } : {}
+        for_each = each.value.network_policy == "isolated" ? { "isolated" : true } : {}
         content {
           pod_selector {}
         }
       }
       dynamic "from" {
-        for_each = each.value == "restricted" ? { "restricted" : true } : {}
+        for_each = each.value.network_policy == "restricted" ? { "restricted" : true } : {}
         content {}
       }
     }
   }
 }
+
+resource "kubernetes_network_policy_v1" "acme" {
+  for_each = {
+    for namespace, project in local.network_policy_per_namespace : namespace => project if project.network_policy != ""
+  }
+
+  metadata {
+    name      = "network-policy-allow-acme"
+    namespace = var.use_existing_kubernetes_namespaces ? data.kubernetes_namespace.namespace[each.key].metadata[0].name : resource.kubernetes_namespace.namespace[each.key].metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {
+        "acme.cert-manager.io/http01-solver" = "true"
+      }
```

Comment:
To be more future proof, could you use a variable for this labels matcher with only this label as default?